src/Entity/EstimatedArrivalTrait.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace Ups\Entity;
+
+trait EstimatedArrivalTrait
+{
+    private $Arrival;
+    private $Pickup;
+    private $DayOfWeek;
+    private $CustomerCenterCutoff;
+    private $DelayCount;
+    private $HolidayCount;
+    private $RestDays;
+    private $TotalTransitDays;
+
+    /**
+     * @param \stdClass|null $response
+     */
+    public function build(\stdClass $response = null)
+    {
+        if (null !== $response) {
+            if (isset($response->Arrival)) {
+                $this->Arrival = new Arrival($response->Arrival);
+            }
+            if (isset($response->Pickup)) {
+                $this->Pickup = new Pickup($response->Pickup);
+            }
+            if (isset($response->HolidayCount)) {
+                $this->HolidayCount = $response->HolidayCount;
+            }
+            if (isset($response->DelayCount)) {
+                $this->DelayCount = $response->DelayCount;
+            }
+            if (isset($response->DayOfWeek)) {
+                $this->DayOfWeek = $response->DayOfWeek;
+            }
+            if (isset($response->TotalTransitDays)) {
+                $this->TotalTransitDays = $response->TotalTransitDays;
+            }
+            if (isset($response->CustomerCenterCutoff)) {
+                $this->CustomerCenterCutoff = $response->CustomerCenterCutoff;
+            }
+            if (isset($response->RestDays)) {
+                $this->RestDays = $response->RestDays;
+            }
+        }
+    }
+
+    /**
+     * @return Arrival
+     */
+    public function getArrival()
+    {
+        return $this->Arrival;
+    }
+
+    /**
+     * @param Arrival $Arrival
+     */
+    public function setArrival($Arrival)
+    {
+        $this->Arrival = $Arrival;
+    }
+
+    /**
+     * @return Pickup
+     */
+    public function getPickup()
+    {
+        return $this->Pickup;
+    }
+
+    /**
+     * @param Pickup $Pickup
+     */
+    public function setPickup($Pickup)
+    {
+        $this->Pickup = $Pickup;
+    }
+
+    /**
+     * @return mixed
```

Comment:
What about these type hints? Might be old, but might also be nice to adjust them to string where applicable.